lib/ws/ws.js

```diff
@@ -9,13 +9,86 @@ var factory = require('./factory');
 var authMethod = 'auth';
 
 /**
+ * race takes an array of promises and is resolved with the first promise
+ * that resolves, or rejects. After the first resolve or rejection, all
+ * future resolutions and rejections will be discarded.
+ * @param {Promise[]} promises
+ * @return {Promise}
+ */
+function race(promises) {
```

Comment:
We do allow the user to inject their own promise via setting BeamSocket.Promise -- that's what's done on the frontend. Using Bluebird would not have removed the need to reinvent this, since the API in the client only requires an A+ promise implementation.